pkg/controller/highperf/highperf_controller.go

```diff
@@ -0,0 +1,245 @@
+package highperf
+
+import (
+	"fmt"
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+
+	networkclientset "k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned"
+	nodeutil "k8s.io/cloud-provider-gcp/pkg/util"
+	"k8s.io/cloud-provider-gcp/providers/gce"
+	controllersmetrics "k8s.io/component-base/metrics/prometheus/controllers"
+
+	networkv1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1"
+	networkv1alpha1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1alpha1"
+)
+
+const (
+	numOfRetries = 5
+)
+
+type Controller struct {
+	gceCloud         *gce.Cloud
+	networkClientset networkclientset.Interface
+
+	nwInformer   cache.SharedIndexInformer
+	gnpInformer  cache.SharedIndexInformer
+	nodeInformer coreinformers.NodeInformer
+
+	queue workqueue.RateLimitingInterface
+}
+
+func NewHighPerfController(
+	gceCloud *gce.Cloud,
+	networkClientset networkclientset.Interface,
+	nwInformer cache.SharedIndexInformer,
+	gnpInformer cache.SharedIndexInformer,
+	nodeInformer coreinformers.NodeInformer) *Controller {
+
+	hc := &Controller{
+		gceCloud:         gceCloud,
+		networkClientset: networkClientset,
+		nwInformer:       nwInformer,
+		gnpInformer:      gnpInformer,
+		nodeInformer:     nodeInformer,
+		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "highperf"),
+	}
+
+	// TODO (bmorettodama): Add event handlers for Network and GNP object DeleteFunc
+
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    nodeutil.CreateAddNodeHandler(func(node *v1.Node) error{
```

Comment:
I thought we moved away from this nodeutil thing in a previous revision, I really prefer to not carryon with this, appreciate if you can add a normal handler.

For the future you'll may need to unit test, so it is better if you add it as a 
```
func (c *Controller)  addNode(obj interface{}) {
			key, err := cache.MetaNamespaceKeyFunc(obj)
			if err != nil {
				klog.Errorf("failed to get node key: %v", err)
				return
			}
			c.queue.Add(key)
```

and add it here as 
```
AddFunc: c.addNode
```

Reply:
Ah I misunderstood the first time around. Fixed now.